From: Bodo Stroesser <bstroesser@fujitsu-siemens.com>
Subject: fix for skas patch
Patch-mainline: never

This patch fixes a bug in SKAS3 host patch v8-rc2 for 2.6.11,
that causes syscalls not to be intercepted, when PTRACE_SYSEMU
is used.

Signed-off-by: Bodo Stroesser <bstroesser@fujitsu-siemens.com>
Acked-by: kraxel@suse.de
---

--- linux-2.6.11-rc4/arch/i386/kernel/ptrace.c.orig	2005-02-15 15:33:26.000000000 +0100
+++ linux-2.6.11-rc4/arch/i386/kernel/ptrace.c	2005-02-15 15:58:18.000000000 +0100
@@ -802,7 +802,7 @@ int do_syscall_trace(struct pt_regs *reg
 	if (is_singlestep)
 		send_sigtrap(current, regs, 0);
 
-	if (!test_thread_flag(TIF_SYSCALL_TRACE))
+	if (!test_thread_flag(TIF_SYSCALL_TRACE) && !is_sysemu)
 		return 0;
 
 	/* the 0x80 provides a way for the tracing parent to distinguish
